### ISSUE TITLE
TypeError: no implicit conversion of nil into Array on FetchGraphWorker

### ISSUE THREAD

**Issue by luisgreg99** [OregonDigital/OD2#1231]
### Descriptive summary

I tried a test migration on latest master branch, and we are getting this error on FetchGraphWorker job

```
DEPRECATION WARNING: Blacklight::Document#[]= is deprecated; use obj.to_h.[]= instead. (called from block (2 levels) in perform at /data/app/workers/fetch_graph_worker.rb:42)
2020-07-02T18:12:08.343Z pid=1 tid=oti03i9rd class=FetchGraphWorker jid=4a9a2e7275ae84eeca170e93 elapsed=24.465 INFO: fail
2020-07-02T18:12:08.343Z pid=1 tid=oti03i9rd WARN: {"context":"Job raised exception","job":{"retry":11,"queue":"default","class":"FetchGraphWorker","args":["fx719n78j","admin@example.org"],"jid":"4a9a2e7275ae84eeca170e93","created_at":1593713500.49423,"enqueued_at":1593713503.8718967},"jobstr":"{\"retry\":11,\"queue\":\"default\",\"class\":\"FetchGraphWorker\",\"args\":[\"fx719n78j\",\"admin@example.org\"],\"jid\":\"4a9a2e7275ae84eeca170e93\",\"created_at\":1593713500.49423,\"enqueued_at\":1593713503.8718967}"}
2020-07-02T18:12:08.344Z pid=1 tid=oti03i9rd WARN: TypeError: no implicit conversion of nil into Array
2020-07-02T18:12:08.356Z pid=1 tid=oti03i9rd WARN: /data/app/workers/fetch_graph_worker.rb:45:in `+'
/data/app/workers/fetch_graph_worker.rb:45:in `block (2 levels) in perform'
/usr/local/bundle/gems/active-triples-1.1.0/lib/active_triples/relation.rb:323:in `block in each'
/usr/local/bundle/gems/rdf-3.1.4/lib/rdf/mixin/enumerable.rb:483:in `block in each_object'
/usr/local/bundle/gems/rdf-3.1.4/lib/rdf/mixin/queryable.rb:299:in `<<'
/usr/local/bundle/gems/rdf-3.1.4/lib/rdf/mixin/queryable.rb:299:in `block (2 levels) in enum_for'
/usr/local/bundle/gems/rdf-3.1.4/lib/rdf/repository.rb:422:in `block (4 levels) in query_pattern'
/usr/local/bundle/gems/hamster-3.0.0/lib/hamster/trie.rb:40:in `block in each'
/usr/local/bundle/gems/hamster-3.0.0/lib/hamster/trie.rb:40:in `each'
/usr/local/bundle/gems/hamster-3.0.0/lib/hamster/trie.rb:40:in `each'
/usr/local/bundle/gems/hamster-3.0.0/lib/hamster/hash.rb:331:in `each'
/usr/local/bundle/gems/rdf-3.1.4/lib/rdf/repository.rb:420:in `block (3 levels) in query_pattern'
/usr/local/bundle/gems/rdf-3.1.4/lib/rdf/repository.rb:419:in `each'
/usr/local/bundle/gems/rdf-3.1.4/lib/rdf/repository.rb:419:in `block (2 levels) in query_pattern'
/usr/local/bundle/gems/rdf-3.1.4/lib/rdf/repository.rb:411:in `each'
/usr/local/bundle/gems/rdf-3.1.4/lib/rdf/repository.rb:411:in `block in query_pattern'
/usr/local/bundle/gems/rdf-3.1.4/lib/rdf/repository.rb:394:in `each'
/usr/local/bundle/gems/rdf-3.1.4/lib/rdf/repository.rb:394:in `query_pattern'
/usr/local/bundle/gems/rdf-3.1.4/lib/rdf/mixin/queryable.rb:299:in `block in enum_for'
/usr/local/bundle/gems/rdf-3.1.4/lib/rdf/mixin/enumerable.rb:160:in `each'
/usr/local/bundle/gems/rdf-3.1.4/lib/rdf/mixin/enumerable.rb:160:in `each'
/usr/local/bundle/gems/rdf-3.1.4/lib/rdf/mixin/enumerable.rb:160:in `each_statement'
/usr/local/bundle/gems/rdf-3.1.4/lib/rdf/mixin/enumerable.rb:479:in `each_object'
/usr/local/bundle/gems/active-triples-1.1.0/lib/active_triples/relation.rb:487:in `objects'
/usr/local/bundle/gems/active-triples-1.1.0/lib/active_triples/relation.rb:321:in `each'
/data/app/workers/fetch_graph_worker.rb:27:in `block in perform'
/data/app/workers/fetch_graph_worker.rb:26:in `each'
/data/app/workers/fetch_graph_worker.rb:26:in `perform'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/processor.rb:196:in `execute_job'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/processor.rb:164:in `block (2 levels) in process'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/middleware/chain.rb:133:in `invoke'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/processor.rb:163:in `block in process'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/processor.rb:136:in `block (6 levels) in dispatch'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/job_retry.rb:111:in `local'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/processor.rb:135:in `block (5 levels) in dispatch'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/rails.rb:43:in `block in call'
/usr/local/bundle/gems/activesupport-5.1.7/lib/active_support/execution_wrapper.rb:85:in `wrap'
/usr/local/bundle/gems/activesupport-5.1.7/lib/active_support/reloader.rb:68:in `block in wrap'
/usr/local/bundle/gems/activesupport-5.1.7/lib/active_support/execution_wrapper.rb:85:in `wrap'
/usr/local/bundle/gems/activesupport-5.1.7/lib/active_support/reloader.rb:67:in `wrap'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/rails.rb:42:in `call'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/processor.rb:131:in `block (4 levels) in dispatch'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/processor.rb:257:in `stats'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/processor.rb:126:in `block (3 levels) in dispatch'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/job_logger.rb:13:in `call'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/processor.rb:125:in `block (2 levels) in dispatch'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/job_retry.rb:78:in `global'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/processor.rb:124:in `block in dispatch'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/logger.rb:10:in `with'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/job_logger.rb:33:in `prepare'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/processor.rb:123:in `dispatch'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/processor.rb:162:in `process'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/processor.rb:78:in `process_one'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/processor.rb:68:in `run'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/util.rb:15:in `watchdog'
/usr/local/bundle/gems/sidekiq-6.0.7/lib/sidekiq/util.rb:24:in `block in safe_thread'
```

### Expected behavior
It looks like it was expecting an array value for `solr_doc['keyword_tesim']` in this line [app/workers/fetch_graph_worker.rb#L45](https://github.com/OregonDigital/OD2/blob/165e83ed2c8a15c2a9a12660c6fbc04ff36f2971/app/workers/fetch_graph_worker.rb#L45) but my test bag didn't have any keyword so it was returning nil.

I think a `to_a` should do it to prevent the error in the queue:

```
-        solr_doc['topic_combined_label_tesim'] = val + solr_doc['keyword_tesim'] if topic_combined_facet?(controlled_prop)
+        solr_doc['topic_combined_label_tesim'] = val + solr_doc['keyword_tesim'].to_a if topic_combined_facet?(controlled_prop)
```

### Related work
https://github.com/OregonDigital/OD2/pull/1224
https://github.com/OregonDigital/OD2/pull/1224/commits/5694aa1f3b1cabd64ffdbfb4ca5061b471c691dc

### Accessibility Concerns

None

**Comment by CGillen** [comment]
QA:
- [ ] Add a work with unique `subject` controlled value and NO keyword metadata
- [ ] Work saves successfully
- [ ] Some time later (around 5 minutes maybe, but no guarantee) `subject` metadata shows up on work show page
- [ ] `Topic` combined facet on search page will facet down to the created work

**Comment by shieldsb** [comment]
QA fail.  On first test, subject metadata showed up on work show page only after the work was edited, and the subject did not appear in the topic facet. On the second test, the subject metadata did not appear on the work show page and was not searchable or in the topic facet. 

First test: https://staging.oregondigital.org/concern/images/j098zb128?locale=en 

Second test: https://staging.oregondigital.org/concern/images/7m01bk74n?locale=en

**Comment by shieldsb** [comment]
QA fail.  Subject metadata still does not appear on the work show page.

**Comment by shieldsb** [comment]
QA fail.  Subject metadata did appear on the work show page, but not in the topic facet. Here is the test work: https://staging.oregondigital.org/concern/images/dn39x158j?locale=en

Also, location metadata did not appear on the work show page during this test.

**Comment by CGillen** [comment]
We're a little out of the woods here and dealing with a seemingly different issue. After RC2, Hyrax added their new pub-sub framework for hooking into callbacks. It seems some of the run order for jobs have changed and one of those deposit jobs kicks off a reindex of the work that may be clobbering some of the asynchronous fetch data.
I think we need to hook the fetchgraphworker after those jobs, which might mean double hooking it for deposit because the hook for update is also fired on deposit

**Comment by CGillen** [comment]
I'm pretty sure the facet not appearing will be fixed with #1279, which Brandon should be working on in an upcoming PR.
The other half, with location not showing up right, might be fixed then as well. I'm going to hold off and wait and see what those fixes bring first.

**Comment by CGillen** [comment]
QA for the original issue has passed and some of these problems has been resolved with other work on the FetchGraphWorker. If we still see some of these bugs we'll open a new ticket
